server/sonar-server/src/main/java/org/sonar/server/computation/ws/ActivityWsAction.java

```diff
@@ -36,15 +38,22 @@
 import org.sonar.db.ce.CeActivityDto;
 import org.sonar.db.ce.CeActivityQuery;
 import org.sonar.db.ce.CeTaskTypes;
+import org.sonar.db.component.ComponentDto;
+import org.sonar.db.component.ComponentDtoFunctions;
+import org.sonar.db.component.ComponentQuery;
+import org.sonar.server.exceptions.BadRequestException;
 import org.sonar.server.exceptions.ForbiddenException;
 import org.sonar.server.user.UserSession;
 import org.sonar.server.ws.WsUtils;
 import org.sonarqube.ws.Common;
 import org.sonarqube.ws.WsCe;
 
+import static java.lang.String.format;
+
 public class ActivityWsAction implements CeWsAction {
 
   private static final String PARAM_COMPONENT_UUID = "componentId";
+  private static final String PARAM_COMPONENT_QUERY = "componentQuery";
```

Comment:
please use the generic way to handle query: an example https://github.com/SonarSource/sonarqube/blob/adb1bccb465775981bb6b6c85c1df4d3ad9191dd/server/sonar-server/src/main/java/org/sonar/server/permission/ws/SearchProjectPermissionsAction.java#L69-69